### PR TITLE
Fix docs mobile layout

### LINKED ISSUE #42921: Mobile layout for the docs website is broken
**This is a Bug Report**

**Problem:**
The mobile screen layout of the [website](https://kubernetes.io/docs/home/) appears to be broken and disrupted. 
Please refer to the below screenshot for a visual representation of the problem.

layout-issue

**Proposed Solution:**
Fix the website  layout for smaller mobile screens

/area web-development

### PR DESCRIPTION
**Pull request by priyanshuchaudhary53**
This PR fixes #42921 and #42947

### CODE REVIEW

File: layouts/partials/css.html [hunk 1]
```diff
@@ -14,6 +14,19 @@
     background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAeCAYAAADkftS9AAAAIklEQVQoU2M4c+bMfxAGAgYYmwGrIIiDjrELjpo5aiZeMwF+yNnOs5KSvgAAAABJRU5ErkJggg==');
     cursor: col-resize;
 }
+
+@media (max-width: 768px) {
+  #sidebarnav,
+  #maindoc {
+    width: 100% !important;
```
Comment (sftim) [inline layouts/partials/css.html:21]: Not ideal. We can merge this, but it's best to avoid `!important` in CSS website styles.

**Review by sftim — COMMENTED** [review]
/lgtm

**Comment by sftim** [comment]
Thanks for the PR @priyanshuchaudhary53.  If you can find a way - even if it means a wider change - to avoid the `!important`, then please do that.

File: layouts/partials/css.html (hunk 1 above)
Comment (priyanshuchaudhary53) [reply layouts/partials/css.html]: Okay, I have updated the JS file and removed `!important` in the new commit. Now, the Split.js splitter is not initialized on mobile screen sizes. Please review the changes.

**Comment by tengqm** [comment]
Thanks.
/approve

**Comment by priyanshuchaudhary53** [comment]
@sftim @tengqm Is there anything else that needs to be done from my end?

**Comment by dipesh-rawat** [comment]
The changes in this PR addresses the issue to some extent but not completely. Several pages (refer to the few links below) still exhibit rendering problems, which might be related to the broader problem mentioned in issue #42947

- https://deploy-preview-42943--kubernetes-io-main-staging.netlify.app//docs/tasks/manage-kubernetes-objects/declarative-config/

- https://deploy-preview-42943--kubernetes-io-main-staging.netlify.app/docs/tasks/manage-kubernetes-objects/imperative-command/

**Comment by tengqm** [comment]
> The changes in this PR addresses the issue to some extent but not completely. Several pages (refer to the few links below) still exhibit rendering problems, which might be related to the broader problem mentioned in issue #42947
> 
>     * https://deploy-preview-42943--kubernetes-io-main-staging.netlify.app//docs/tasks/manage-kubernetes-objects/declarative-config/
> 
>     * https://deploy-preview-42943--kubernetes-io-main-staging.netlify.app/docs/tasks/manage-kubernetes-objects/imperative-command/

The right hand navigation bar, in my opinion, is useless. If we ever need a second navigation menu in addition to the left one, it may mean we may have got a long page to be split.
Huge pages are not easy to read, and not easy to maintain.

In addition to that, the built-in buttons (Edit, File issue .. etc) are useless as well.

**Comment by tengqm** [comment]
With this merged, we can proceed to resolve the in-page navigation issue.

**Comment by priyanshuchaudhary53** [comment]
> The changes in this PR addresses the issue to some extent but not completely. Several pages (refer to the few links below) still exhibit rendering problems, which might be related to the broader problem mentioned in issue #42947
> 
> * https://deploy-preview-42943--kubernetes-io-main-staging.netlify.app//docs/tasks/manage-kubernetes-objects/declarative-config/
> * https://deploy-preview-42943--kubernetes-io-main-staging.netlify.app/docs/tasks/manage-kubernetes-objects/imperative-command/

This issue can easily be fixed by adding CSS `max-width: 100%`  on `#sidebarnav` and `#maindoc` divs

**Comment by priyanshuchaudhary53** [comment]
> > The changes in this PR addresses the issue to some extent but not completely. Several pages (refer to the few links below) still exhibit rendering problems, which might be related to the broader problem mentioned in issue #42947
> > 
> > * https://deploy-preview-42943--kubernetes-io-main-staging.netlify.app//docs/tasks/manage-kubernetes-objects/declarative-config/
> > * https://deploy-preview-42943--kubernetes-io-main-staging.netlify.app/docs/tasks/manage-kubernetes-objects/imperative-command/
> 
> This issue can easily be fixed by adding CSS `max-width: 100%` on `#sidebarnav` and `#maindoc` divs

Fixed the above-stated issue and also #42947 in the recent commit

**Comment by reylejano** [comment]
Thank you for the fix
/lgtm

**Pull request merged**